esx_service/utils/auth.py

```diff
@@ -326,6 +326,25 @@ def add_volume_to_volumes_table(tenant_uuid, datastore, vol_name, vol_size_in_MB
 
     return None
 
+def remove_volume_from_volumes_table(tenant_uuid, datastore, vol_name):
+    """ Remove volume from volumes table. """
+    _auth_mgr = get_auth_mgr()
+
+    logging.debug("remove volumes from volumes table(%s %s %s)", tenant_uuid, datastore,
+                  vol_name)
+    try:
+        _auth_mgr.conn.execute(
+                    "DELETE FROM volumes WHERE tenant_id = ? AND datastore = ? AND volume_name = ?", 
+                    [tenant_uuid, datastore, vol_name]
+            )
+        _auth_mgr.conn.commit()
```

Comment:
It looks like we have quite some duplicate code in this module. We should refactor the code to add the template steps in a util class. The util class should expose some common DB operation methods to handle: 1) get connection; 2) execute statement; 3) commit (for CUD only); 4) error handling; 5) result processing (pagination support, etc.) - for R (query) only

I'm not suggesting to address this in the current review, but we should file a bug to track this improvement.